config/routes.rb

```diff
@@ -1,21 +1,57 @@
 require 'sidekiq/web' if defined? Sidekiq
 require Rails.root.join('lib', 'router_constraints.rb')
 
-Ontohub::Application.routes.draw do
+Specroutes.define(Ontohub::Application.routes) do
 
   resources :filetypes, only: :create
 
   # IRI Routing #
   ###############
   #
+  # as per Loc/Id definition
+
+  ontology_subsites = %i(
+    mappings symbols children
+    sentences theorems comments
+    metadata ontology_versions graphs
+    projects categories tasks license_models formality_levels
+  )
+
+  ontology_subsites.each do |category|
+    specified_get "/:repository_id/*locid///#{category}" => "#{category}#index",
+      as: :"ontology_iri_#{category}",
+      constraints: [
+        LocIdRouterConstraint.new(Ontology, ontology: :ontology_id),
+      ]
+  end
 
-  get ':repository_id(/*path)/:file',
-    controller:  :ontologies,
-    action:      :show,
-    as:          :ontology_iri,
-    constraints: GroupedConstraint.new(
-      IRIRouterConstraint.new,
-      MIMERouterConstraint.new('text/plain', 'text/html'))
+  specified_get '/:repository_id/*locid' => 'ontologies#show',
+    as: :ontology_iri,
+    constraints: [
+      LocIdRouterConstraint.new(Ontology, ontology: :id),
+      MIMERouterConstraint.new('text/plain', 'text/html'),
+    ]
+
+  specified_get '/:repository_id/*locid' => 'mappings#show',
+    as: :mapping_iri,
+    constraints: [
+      LocIdRouterConstraint.new(Mapping, ontology: :ontology_id, element: :id),
+      MIMERouterConstraint.new('text/plain', 'text/html'),
+    ]
+
+  specified_get '/:repository_id/*locid' => 'symbols#index',
+    as: :symbol_iri,
```

Comment:
Style/AlignHash: Align the elements of a hash literal if they span more than one line.